lib/usd/utils/util.cpp

```diff
@@ -71,6 +71,49 @@ namespace
             {"nodePath", arc.GetTargetNode().GetPath().GetString()},
         };
     }
+
+    void replaceReferenceItems(const UsdPrim& oldPrim, 
+                               const SdfPath& newPath,
+                               const SdfReferencesProxy& referencesList,
+                               SdfListOpType op)
+    {
+        // set the itemVector based on the SdfListOpType
+        auto itemVector = referencesList.GetAppendedItems();
+        if (op == SdfListOpTypePrepended) {
+            itemVector = referencesList.GetPrependedItems();
+        } else if (op == SdfListOpTypeOrdered) {
+            itemVector = referencesList.GetOrderedItems();
+        } else if (op == SdfListOpTypeAdded) {
+            itemVector = referencesList.GetAddedItems();
+        } else if (op == SdfListOpTypeDeleted) {
+            itemVector = referencesList.GetDeletedItems();
+        }
+
+        // fetching the existing SdfReference items and using 
+        // the Replace() method to replace them with updated SdfReference items.
+        for (const SdfReference &ref : itemVector)
+        {
+            if (ref.IsInternal())
+            {
+                SdfPath finalPath;
+                if(oldPrim.GetPath() == ref.GetPrimPath()) {
+                    finalPath = newPath;
+                }
+                else if(ref.GetPrimPath().HasPrefix(oldPrim.GetPath())) {
+                    finalPath = ref.GetPrimPath().ReplacePrefix(oldPrim.GetPath(), newPath);
+                }
+
+                if(finalPath.IsEmpty()) {
+                    continue;
+                }
+
+                // replace the old reference with new one
+                SdfReference newRef;
+                newRef.SetPrimPath(finalPath);
+                itemVector.Replace(ref, newRef);
+            }
+        }
+    }
 }
```

Comment:
@ppt-adsk Per conversation with Spiff, it was suggested that there would be a more efficient technique to approach how the internal references list ops can be updated.

In the new approach we directly replacing the old SdfReference with the new SdfReference and call the itemVector Replace() to switch them.  This is more efficient than using the Usd API approach where I needed to clear the list first and add the new one reference.